### PR TITLE
add svelte support to webpack encore

### PR DESCRIPTION
**Pull request by zairigimad**
> Svelte is a radical new approach to building user interfaces. Whereas traditional frameworks like React and Vue do the bulk of their work in the browser, Svelte shifts that work into a compile step that happens when you build your app. [https://svelte.dev/](https://svelte.dev/)

The goal of this PR is to add support of Svelte to Webpack Encore.

- [ ] Update Documentation https://symfony.com/doc/current/frontend/encore/advanced-config.html#having-the-full-control-on-loaders-rules

### CODE REVIEW

File: lib/config-generator.js [hunk 1]
```diff
@@ -399,6 +399,17 @@ class ConfigGenerator {
             }));
         }
 
+        if (this.webpackConfig.useSvelte) {
+            rules.push(applyRuleConfigurationCallback('svelte', {
```
Comment (Kocal) [inline lib/config-generator.js:403]: Would require to update https://symfony.com/doc/current/frontend/encore/advanced-config.html#having-the-full-control-on-loaders-rules, thanks!

Comment (zairigimad) [reply lib/config-generator.js:403]: yes after the merge :)

**Comment by Rodrigo001-de** [comment]
Project made with TypeScript, react.Js, React-native. Recycling day design

https://github.com/Rodrigo001-de/Projeto-Ecoleta-advanced

**Comment by Volubyl** [comment]
Hey!

Do you need a hand with that PR?

It would be great to have this feature in the webpack!

If I can help in any way, let me know!

**Comment by zairigimad** [comment]
@Volubyl yes it will be great if you can help :)

**Comment by Volubyl** [comment]
> @Volubyl yes it will be great if you can help :)

Nice!

Can you tell me what remains to be done with this PR and how I can help?

**Comment by zairigimad** [comment]
> > @Volubyl yes it will be great if you can help :)
> 
> Nice!
> 
> Can you tell me what remains to be done with this PR and how I can help?

when I've added this, it was working, maybe if you can test it and see if something missing or if you have and idea you can add it :)

**Review by weaverryan — COMMENTED** [review]
Sorry for letting this sit for SO long without any attention :/.

The only thing I can see that this is missing is a test in `functional.js`. It would probably include adding a "hello world" type of svelte file into the fixtures, then booting up an Encore setup, executing Encore and asserting that the result works. We have plenty of examples of that sort of thing in that `functional.js` file - React or Vue might be especially good examples to pull from.

Cheers!

**Comment by zairigimad** [comment]
> Sorry for letting this sit for SO long without any attention :/.
> 
> The only thing I can see that this is missing is a test in `functional.js`. It would probably include adding a "hello world" type of svelte file into the fixtures, then booting up an Encore setup, executing Encore and asserting that the result works. We have plenty of examples of that sort of thing in that `functional.js` file - React or Vue might be especially good examples to pull from.
> 
> Cheers!

no problem, I will make the test ASAP.

**Comment by zairigimad** [comment]
Hey @weaverryan I've added the functional test for svelte :)

**Review by AbdelilahJabri — APPROVED** [review]
Nicely done, thank you for this beautiful support

**Comment by CrazyHackGUT** [comment]
Any chances to be merged in near future?

**Comment by weaverryan** [comment]
Thank you @zairigimad! I'm so sorry - and embarrassed by how much this fell off my radar. Apologies for the insane delay, but this is excellent work!

**Comment by zairigimad** [comment]
> Thank you @zairigimad! I'm so sorry - and embarrassed by how much this fell off my radar. Apologies for the insane delay, but this is excellent work!

No problem you are doing an incredible work maintaining lot of things in Symfony ecosystem, thanks to you. I’m very happy that it’s finally merged 🎉

**Pull request merged**